setup.go

```diff
@@ -74,47 +85,34 @@ func parse(c *caddy.Controller) (*Gateway, error) {
 		}
 
 		for c.NextBlock() {
+			args := c.RemainingArgs()
+			if len(args) == 0 {
+				return nil, c.ArgErr()
+			}
 			switch c.Val() {
 			case "resources":
-				args := c.RemainingArgs()
-
 				gw.updateResources(args)
 
 				if len(args) == 0 {
 					return nil, c.Errf("Incorrectly formated 'resource' parameter")
 				}
```

Comment:
thx for refactoring @k0da . The code is quickly understandable now 👍 

Can you ensure we need this code block ? Seems it can't occur and if it does, it exists anyway.

Reply:
Yes, it wasn't needed